multiuser/keycloak/che-multiuser-keycloak-user-remover/src/main/java/org/eclipse/che/multiuser/keycloak/server/KeycloakUserRemover.java

```diff
@@ -0,0 +1,202 @@
+        throw new ConfigurationException("Keycloak administrator username or password not set.");
+      }
+      this.keycloakTokenEndpoint = serverUrl + "/realms/master/protocol/openid-connect/token";
+      this.keycloakRemoveUserUrl = serverUrl + "/admin/realms/" + realm + "/users/";
+    }
+  }
+
+  /**
+   * Remove user from Keycloak server by given user id.
+   *
+   * @param userId the user if to remove
+   * @throws ServerException when can't remove user from Keycloak
+   */
+  public void removeUserFromKeycloak(String userId) throws ServerException {
+    try {
+      String token = requestToken();
+      int responseCode =
+          requestFactory
+              .fromUrl(keycloakRemoveUserUrl + userId)
+              .setAuthorizationHeader("Bearer " + token)
+              .useDeleteMethod()
+              .request()
+              .getResponseCode();
+      if (responseCode != 204) {
+        throw new ServerException("Can't remove user from KeyClock. UserId:" + userId);
+      }
+    } catch (IOException | ApiException e) {
+      LOG.warn("Exception during removing user from Keycloak", e);
+      throw new ServerException("Exception during removing user from Keycloak", e);
+    }
+  }
+
+  private String requestToken() throws ServerException {
+    String accessToken;
+    HttpURLConnection http = null;
+    try {
+      http = (HttpURLConnection) new URL(keycloakTokenEndpoint).openConnection();
+      http.setConnectTimeout(60000);
+      http.setReadTimeout(60000);
+      http.setRequestMethod(POST);
+      http.setAllowUserInteraction(false);
+      http.setRequestProperty(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_FORM_URLENCODED);
+      http.setInstanceFollowRedirects(true);
+      http.setDoOutput(true);
+      StringBuilder sb = new StringBuilder();
+      sb.append("grant_type=password")
+          .append("&username=")
+          .append(keycloakUser)
+          .append("&password=")
+          .append(keycloakPassword)
+          .append("&client_id=admin-cli");
+      try (OutputStream output = http.getOutputStream()) {
+        output.write(sb.toString().getBytes(UTF_8));
+      }
+      if (http.getResponseCode() != 200) {
+        throw new RuntimeException(
+            "Cannot get access KeyCloak token. Server response code: "
+                + keycloakTokenEndpoint
+                + " "
+                + http.getResponseCode()
+                + IoUtil.readStream(http.getErrorStream()));
+      }
+      final BufferedReader response =
+          new BufferedReader(new InputStreamReader(http.getInputStream(), UTF_8));
+
+      JsonParser jsonParser = new JsonParser();
+      JsonElement jsonElement = jsonParser.parse(response);
+      JsonObject asJsonObject = jsonElement.getAsJsonObject();
+      if (asJsonObject.has("access_token")) {
+        accessToken = asJsonObject.get("access_token").getAsString();
+      } else {
+        LOG.error("Cannot get access KeyCloak token from server response");
```

Comment:
Why you need that? In any failure, you're supposed to log error here https://github.com/eclipse/che/pull/17858/files#diff-6faa75cae54e91254f0b4f0d3d8b9b8bR115